src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -48,9 +49,27 @@ public static class SqlServerBaseRegistrationExtensions
             services.TryAddSingleton(p => p.GetRequiredService<IScriptProvider>() as ScriptProvider<TSchemaVersionEnum>);
             services.TryAddSingleton(p => p.GetRequiredService<IBaseScriptProvider>() as BaseScriptProvider);
             services.TryAddSingleton(p => p.GetRequiredService<ISchemaManagerDataStore>() as SchemaManagerDataStore);
-            services.TryAddSingleton(p => p.GetRequiredService<IPollyRetryLoggerFactory>() as PollyRetryLoggerFactory);
             services.TryAddSingleton(p => p.GetRequiredService<SqlCommandWrapperFactory>() as RetrySqlCommandWrapperFactory);
-            services.TryAddSingleton(p => p.GetRequiredService<ISqlServerTransientFaultRetryPolicyFactory>() as SqlServerTransientFaultRetryPolicyFactory);
+            services.TryAddSingleton<SqlRetryLogicBaseProvider>(p =>
+            {
+                SqlServerDataStoreConfiguration config = p.GetRequiredService<IOptions<SqlServerDataStoreConfiguration>>().Value;
+
+                // using the default transient error codes here https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/SqlConfigurableRetryFactory.cs
+                var options = new SqlRetryLogicOption()
+                {
+                    // Tries 5 times before throwing an exception
+                    NumberOfTries = config.TransientFaultRetryPolicy.ConnectRetryCount,
+
+                    // Preferred gap time to delay before retry
+                    DeltaTime = TimeSpan.FromSeconds(config.TransientFaultRetryPolicy.ConnectRetryIntervalInSeconds),
+
+                    // Maximum gap time for each delay time before retry
+                    MaxTimeInterval = TimeSpan.FromSeconds(config.TransientFaultRetryPolicy.ConnectMaxTimeIntervalInSeconds),
+                };
+
+                // Create a retry logic provider
+                return SqlConfigurableRetryFactory.CreateExponentialRetryProvider(options);
```

Comment:
We could also introduce a new configuration for these values to avoid having to map these values ourselves. Something like:

```csharp
public class SqlClientRetryOptions
{
    public SqlRetryStrategy Strategy { get; set; } // Maps to the various factory methods like None, Exponential, Incremental, and Fixed
    public SqlRetryLogicOption Settings { get; set; } // Unfortunately this is a sealed class! Otherwise, I would say let's extend it
}
```

Then we could allow users to configure these options themselves based on hard-coded values or a config via:
```csharp
services.Configure(x =>
{
    x.NumberOfTries = 12;
});

// OR

services
    .AddOptions<SqlClientRetryOptions>()
    .Bind(config.GetSection("SqlServer:Retry"));
```

With this class, we can update the retry factory to generate a `SqlRetryLogicBaseProvider` in its ctor based on these options:

```csharp
public RetrySqlCommandWrapperFactory(IOptions<SqlClientRetryOptions> options)
{
    SqlClientRetryOptions o = EnsureArg.IsNotNull(options?.Value, nameof(options));
    _retryProvider = o.Strategy switch
    {
        SqlRetryStrategy.None => SqlConfigurableRetryFactory.CreateNoneRetryProvider(),
        SqlRetryStrategy.Fixed => SqlConfigurableRetryFactory.CreateFixedRetryProvider(o.Settings),
        SqlRetryStrategy.Incremental => SqlConfigurableRetryFactory.CreateIncrementalRetryProvider(o.Settings),
        SqlRetryStrategy.Exponential => SqlConfigurableRetryFactory.CreateExponentialRetryProvider(o.Settings),
        _ => throw new ArgumentOutOfRangeException(nameof(options)),
    };
}
```

Reply:
Thanks Will. I like the pattern of reusing the same objects. I also want to be more prescreptive in the framework. All the clients of the SDK should inherit a default retry with an option to override. Mostly to help pass best practices across all clients.